pretty_midi/instrument.py

```diff
@@ -465,7 +465,10 @@ def fluidsynth(self, fs=44100, synthesizer=None, sfid=0, sf2_path=None):
             return np.array([])
 
         # Create a fluidsynth instance if one wasn't provided
+        # Raises if sample rates do not match
         synthesizer, sfid, delete_synthesizer = get_fluidsynth_instance(synthesizer, sfid, fs)
+        # Sample rate required below to compute note positions
+        fs = fs or synthesizer.get_setting('synth.sample-rate')
```

Comment:
I think you can just use synthesizer.get_setting('synth.sample-rate') in either case here, right? Because if the synthesizer is created, it will have the sample rate provided to get_fluidsynth_instance